### PR TITLE
Allow Kerberos authentication with service ticket from CCACHE

### LINKED ISSUE #14: Kerberos authentication with service ticket from ccache
It is currently not possible to authenticate using Kerberos without any credentials other than a ccache file with the service ticket, even though the service ticket is sufficient for authentication.

This issue stems from the fact that [gokrb5](https://github.com/jcmturner/gokrb5) requires a TGT to be present in the ccache file in order instantiate a client (https://github.com/jcmturner/gokrb5/blob/855dbc707a37a21467aef6c0245fcf3328dc39ed/v8/client/client.go#L79-L82).

### PR DESCRIPTION
**Pull request by rtpt-erikgeiser**
This PR fixes #14.

In this PR, I allow the Kerberos client not to hold any credentials if a ccache file with at least one ticket is present. If this is the case, service ticket will later only be fetched from the ccache file because the client can authenticate on its own anyways.

I know this is a bit hacky and should technically probably be fixed in [gokrb5](https://github.com/jcmturner/gokrb5) but the maintainer seems to be inactive and `go-msrpc` already contains some  workarounds around issues with `gokrb5` (like this one https://github.com/oiweiwei/go-msrpc/blob/main/ssp/krb5/keytab.go).

### CODE REVIEW

File: ssp/krb5/authentifier.go [hunk 1]
```diff
@@ -178,21 +198,44 @@ func (a *Authentifier) affirmLogin(ctx context.Context) error {
...
 	if err != nil {
 		return nil, fmt.Errorf("krb5: init: apreq: make client: %w", err)
 	}
 
 	a.client = cli
 
-	if err := a.affirmLogin(ctx); err != nil {
-		return nil, fmt.Errorf("krb5: init: apreq: affirm login: %w", err)
-	}
+	var (
```
Comment (oiweiwei) [inline ssp/krb5/authentifier.go:208]: could you please hide the logic for ccacheWithoutTGT inside a.affirmLogin and a.getServiceTicket?

```
func (a *Authentifier) affirmLogin(...) {
    if ccacheWithoutTGT != nil {
        return nil
    }
    ...
}
```

func (a *Authentifier) getServiceTicket(sname string) (tkt, key, error) {

    if a.ccacheWithoutTGT != nil {
        c, ok := a.ccacheWithoutTGT.GetEntry(...)
        ...
        return tkt, key, nil
     }
     return a.client.GetServiceTicket(sname)
}
```

File: ssp/krb5/authentifier.go [hunk 2]
```diff
@@ -34,6 +37,8 @@ type Authentifier struct {
 	SessionKey types.EncryptionKey
 	// key.
 	state *SecurityService
+
+	ccacheWithoutTGT *credentials.CCache
```
Comment (oiweiwei) [inline ssp/krb5/authentifier.go:41]: put some generic comment on the use-case here as well.

**Comment by oiweiwei** [comment]
perhaps, one day I will have to move to this fork: https://github.com/jfjallid/gokrb5/tree/master/v8/client

**Comment by rtpt-erikgeiser** [comment]
Thanks for the fast review, I implemented the suggestions.

**Review by oiweiwei — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**